test/github-package.test.js

```diff
@@ -8,617 +8,836 @@ import {fileExists, getTempDir} from '../lib/helpers';
-      assert.isTrue(context.isPresent());
+    afterEach(async function() {
+      await githubPackage.deactivate();
 
-      assert.strictEqual(context.getRepository(), githubPackage.getActiveRepository());
-      assert.strictEqual(context.getResolutionProgress(), githubPackage.getActiveResolutionProgress());
-      assert.equal(githubPackage.getActiveWorkdir(), workdirPath);
+      atomEnv.destroy();
     });
 
-    it('uses an active model from a preexisting active pane item', async function() {
-      const [workdirPath1, workdirPath2] = await Promise.all([
-        cloneRepository('three-files'),
-        cloneRepository('three-files'),
-      ]);
-      project.setPaths([workdirPath1, workdirPath2]);
-      await workspace.open(path.join(workdirPath2, 'a.txt'));
-
-      await contextUpdateAfter(() => githubPackage.activate());
+    context('with no project, state, or active pane', function() {
+      beforeEach(async function() {
+        await contextUpdateAfter(githubPackage, () => githubPackage.activate());
+      });
 
-      const context = contextPool.getContext(workdirPath2);
-      assert.isTrue(context.isPresent());
-      assert.strictEqual(context.getRepository(), githubPackage.getActiveRepository());
-      assert.strictEqual(context.getResolutionProgress(), githubPackage.getActiveResolutionProgress());
-      assert.equal(githubPackage.getActiveWorkdir(), workdirPath2);
+      it('uses an undetermined repository context', function() {
+        assert.isTrue(githubPackage.getActiveRepository().isUndetermined());
+      });
     });
 
-    it('uses an active model from serialized state', async function() {
-      const [workdirPath1, workdirPath2, workdirPath3] = await Promise.all([
-        cloneRepository('three-files'),
-        cloneRepository('three-files'),
-        cloneRepository('three-files'),
-      ]);
-      project.setPaths([workdirPath1, workdirPath2, workdirPath3]);
+    context('with only 1 project', function() {
+      let workdirPath;
+      beforeEach(async function() {
+        workdirPath = await cloneRepository('three-files');
+        project.setPaths([workdirPath]);
 
-      await contextUpdateAfter(() => githubPackage.activate({
-        activeRepositoryPath: workdirPath2,
-      }));
+        await contextUpdateAfter(githubPackage, () => githubPackage.activate());
+      });
 
-      const context = contextPool.getContext(workdirPath2);
-      assert.isTrue(context.isPresent());
-      assert.strictEqual(context.getRepository(), githubPackage.getActiveRepository());
-      assert.strictEqual(context.getResolutionProgress(), githubPackage.getActiveResolutionProgress());
-      assert.equal(githubPackage.getActiveWorkdir(), workdirPath2);
+      it('uses the project\'s context', function() {
+        const context = contextPool.getContext(workdirPath);
+        assert.isTrue(context.isPresent());
+        assert.equal(githubPackage.getActiveWorkdir(), workdirPath);
```

Comment:
I see this is just lifted from the previous code right now, but I tend to assert with `.strictEqual` instead of `.equal` unless I have a good reason not to... `.strictEqual` compares its arguments with `===` while `.equal` compares them with `==`, so it has similar properties of avoiding surprising type coercions.

Reply:
will look for this and replace